sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Canonicalize.scala

```diff
@@ -36,15 +36,16 @@ import org.apache.spark.sql.catalyst.rules._
 object Canonicalize extends RuleExecutor[Expression] {
   override protected def batches: Seq[Batch] =
     Batch(
-      "Expression Canonicalization", FixedPoint(100),
+      "Expression Canonicalization", Once,
```

Comment:
Its kind of confusing for this to be a RuleExecutor if we are only going to run it once and its only doing matches.  Should we just move this into `Expression`?